soci/soci_index.go

```diff
@@ -448,10 +445,11 @@ func (b *IndexBuilder) buildSociLayer(ctx context.Context, desc ocispec.Descript
 }
 
 // NewIndex returns a new index.
-func NewIndex(blobs []ocispec.Descriptor, subject *ocispec.Descriptor, annotations map[string]string, legacy bool) *Index {
-	mediaType := OCIArtifactManifestMediaType
-	if legacy {
-		mediaType = ocispec.MediaTypeImageManifest
+func NewIndex(blobs []ocispec.Descriptor, subject *ocispec.Descriptor, annotations map[string]string, artifact bool) *Index {
+	mediaType := ocispec.MediaTypeImageManifest
+
+	if artifact {
+		mediaType = OCIArtifactManifestMediaType
```

Comment:
Since we're using [`image-spec v1.1.0-rc2`](https://github.com/awslabs/soci-snapshotter/blob/7f0c3aa79b204aeb36a70bec40a051bda21c308a/go.mod#L21) and `MediaTypeArtifactManifest` exists in [image-spec](https://github.com/opencontainers/image-spec/blob/19a74bcb54ba211005a68d85c6b359c2947721ce/specs-go/v1/mediatype.go#LL59C2-L59C27), I think we should remove our copy and use `ocispec.MediaTypeArtifactManifest`.

WDYT @Kern--

Reply:
We can. I'm not too rushed because it's not clear that artifact manifests will survive until the release of `v1.1.0`